deploy_cluster.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/sh
+
+SSH_KEY=~/.ssh/id_rsa
+REQUIRED_PKGS="libvirt podman virt-install"
+OCP_VENV=ocp-venv
+
+chk_user()
+{
+	if [ $USER != root ]; then
+		echo "root user needed, current user is $USER"
+		exit -1
+	fi
+}
+
+chk_install_pkgs()
+{
+	local pkgs=""
+	local p
+
+	# rhel and fedora have different names for the qemu package
+	if [ -f /etc/fedora-release ]; then
+		REQUIRED_PKGS="$REQUIRED_PKGS qemu"
+	elif [ -f /etc/redhat-release ]; then
+		REQUIRED_PKGS="$REQUIRED_PKGS qemu-kvm"
+	else
+		echo "Unsupported host, only fedora and redhat distro are supported"
+		exit -1
+	fi
+
+	for P in $REQUIRED_PKGS; do
+		rpm -qv $P >/dev/null || pkgs="$pkgs $P"
+	done
+	if [ -n "$pkgs" ]; then
+		echo "Installing required packages $pkgs"
+		dnf install -y $pkgs
+	fi
+}
+
+chk_create_key()
+{
+	# recent fedura deprecates rsa, but the deply infra hard-codes rsa usage
+	if [ ! -f $SSH_KEY ]; then
+		echo "Generating ssh key..."
+		ssh-keygen -t rsa -a 64 -N '' -f $SSH_KEY
+	fi
+}
+
+chk_update_libvirt_conf()
+{
+	if grep -qe '#\(user\|group\)' /etc/libvirt/qemu.conf; then
+		echo "Reconfiguring qemu"
+		sed -e 's/#\(user\|group\) = ".*"$/\1 = "root"/' -i /etc/libvirt/qemu.conf
+		systemctl restart libvirtd
+	else
+		systemctl start libvirtd
+	fi
+}
+
+chk_create_enter_env() {
+	local install_req=false
+
+	if [ ! -d $OCP_VENV ]; then
+		echo "Creating virtual env $OCP_VENV"
+		python3 -m venv $OCP_VENV
+		install_req=true
+	fi
+
+	source $OCP_VENV/bin/activate
+	if [ "$install_req" = true ]; then
+		pip install --upgrade pip
+		pip install -r requirements.txt
+	fi
+}
+
+chk_user
+chk_create_key
+chk_install_pkgs
+chk_update_libvirt_conf
+chk_create_enter_env
```

Comment:
Let's move these checks inside the python scripts

Reply:
The above checks also includes pkg management/system configuration "actions" - pkg install, conf file changes, etc.. I think the latter should stay outside the python scripts. Do you mean do move even the "action" part in python? 

Moving the checks only would cause code duplication, as we will still need to perform the check here to avoid unconditionally doing slow/unneeded actions here.

Reply:
Since the goal is to support, for example, remove VMs too, we will need to check remote systems too for the right dependencies. For

Reply:
An additional reason is that the dependencies depend on the configuration. For example, if we create VMs locally, then we need qemu, but if we don't, then we don't need qemu. You only know that after loading the config file.